keps/sig-network/20200103-mixed-protocol-lb.md

```diff
@@ -340,17 +341,80 @@ Con:
+ - the CPI implementations shall be prepared to deal with Services with mixed protocol configurations. Either via supporting such Service definitions, or clearly indicating to the users that the Service could not be processed as specified. As we can see from our analysis some CPIs support other protocols than TCP and UDP in the Service definitions, while others support only TCP and UDP. That is the term "mixed protocol support" does not always mean that all possible protocol values are supported by a CPI. For this reason a nicely behaving CPI shall 
+ - indicate clearly to the user what ports with what protocols have been opened on the LB
+ - preferably not create any Cloud LB resources if the Service definition contains unsupported protocols.
+ 
+ In order to provide a way for the CPI to indicate port statuses to the user we would add the following new `portStatus` list of port statuses to `Service.status.loadBalancer`, so the CPI can indicate the status of the LB:
+
+```json
+"io.k8s.api.core.v1.PortStatus": {
+  "description": "PortStatus contains details for the current status of this port.",
+  "properties": {
+    "port": {
+      "description": "Port number",
+      "format": "int32",
+      "type": "integer"
+    },
+    "protocol": {
+      "description": "The protocol for this port",
+      "type": "string"
+    },
+    "name": {
+      "description": "The name of this port within the service.",
+      "type": "string"
+    },
+    "ready": {
+      "description": "Specifies whether the port was configured for the load-balancer.",
+      "type": "boolean"
+    },
+    "message": {
+      "description": "A human readable message indicating details about why the port is in this condition.",
+      "type": "string"
+    }
+  },
+  "type": "object"
+},
+"io.k8s.api.core.v1.LoadBalancerStatus": {
+  "description": "LoadBalancerStatus represents the status of a load-balancer.",
+  "properties": {
+    "ingress": {
+      "description": "Ingress is a list containing ingress points for the load-balancer. Traffic intended for the service ld be sent to these ingress points.",
+      "items": {
+        "$ref": "#/definitions/io.k8s.api.core.v1.LoadBalancerIngress"
+      },
+      "type": "array"
+    },
+    "portStatuses": {
```

Comment:
I think we want `ports` under `ingress` - ingress is a plural on purpose, so ports should be tracked based on each ingress point.

Reply:
Fixed, moved the portStatuses under ingress.